lib/matplotlib/axes/_axes.py

```diff
@@ -2172,23 +2192,25 @@ def bar(self, x, height, width=0.8, bottom=None, *, align="center",
         else:
             raise ValueError('invalid orientation: %s' % orientation)
 
+        x, height, width, y, linewidth = np.broadcast_arrays(
```

Comment:
Should this go below unit conversion?  Otherwise I think units will be dropped if width is a list of unitized stuff?

Reply:
This will drop stuff like pint units or astropy units.
I remember some discussion a while ago which basically reached the conclusion that we cannot call *any* numpy function before unit handling (tbh I was rather unhappy about that, but the goal is not to relitigate this here), perhaps @dopplershift will remember this better.

Reply:
Can someone provide an example?   This works fine for all the categorical, unit, and date tests.  If we are supporting something that doesn’t survive broadcasting we need to start testing it.  

I’m not sure why we broadcast here at all anyway.

Reply:
AFAICS, broadcasting is used to support scalar parameters such as `plt.bar([1, 2, 3], [3,5,2], width=0.5)`

Reply:
For the record, `broadcast_arrays` **will** drop pint units--it might possibly work with astropy since it's an ndarray subclass. Broadcasting itself works fine, it's calling functions that need to allocate new arrays that's the problem; there are currently no hooks to allow overriding what type to use when allocating an empty array.

Reply:
Agreed - sorry, should have responded here - I did move the broadcast after the convert...